### ISSUE TITLE
Provide a way to determine wether a timezone is deprecated

### ISSUE THREAD

**Issue by Schnark** [moment/moment-timezone#956]
In some cases it would be useful to known wether a timezone is deprecated or not, but currently moment-timezone does not provide this information.
For example, you might want to provide the user with a list of all timezones to chose from. If you use `moment.tz.names()` you will get all timezones, including some old deprecated ones.
So it would be nice if you could provide that information for each zone, or alternatively add a parameter to `moment.tz.names` to restrict the output to non-deprecated zones only.
The best current workaround seems to be to get the list of countries a zone is used in. If it isn't empty the zone is not deprecated. But the reverse isn't true in some special cases (especially `Etc/*`).

**Comment by gilmoreorless** [comment]
This sort of discussion has come up a few times in the IANA tzdb mailing list. There have been [some proposals](https://mm.icann.org/pipermail/tz/2021-November/031074.html) but nothing concrete yet, because there are differing opinions about what exactly counts as "deprecated".

**Comment by ichernev** [comment]
Yep, moment-timezone is a "dumb" consumer of IANA, and it doesn't even export everyting IANA has. If you really need such information you should source it yourself (given that even IANA doesn't have it ATM). Dealing with all the people in the world is a complicated business ;-)
